src/app/core/utils/api-token/api-token.service.ts

```diff
@@ -62,121 +63,140 @@ const DEFAULT_EXPIRY_TIME = 3600000;
-            )
-          )
+      this.cookieChangeEvent$ = this.checkCookieRoutine().pipe(shareReplay(1));
+      this.saveTokenRoutine();
+      this.logoutRoutine();
+      this.tokenVanishRoutine();
+      this.tokenCreatedOnAnotherTabRoutine();
+    }
+  }
+
+  private checkCookieRoutine() {
+    return this.appRef.isStable.pipe(
+      whenTruthy(),
+      first(),
+      mergeMap(() =>
+        interval(1000).pipe(
+          map(() => this.parseCookie()),
+          pairwise(),
+          distinctUntilChanged((prev, curr) => isEqual(prev, curr))
         )
-        .subscribe(type => {
-          this.apiToken$.next(undefined);
-          this.cookieVanishes$.next(type);
-        });
-
-      // cookie vanishes routine when user is logged out in an another tab
-      appRef.isStable
-        .pipe(
-          whenTruthy(),
-          first(),
-          mergeMap(() =>
-            interval(1000).pipe(
-              map(() => this.parseCookie()),
-              pairwise(),
-              filter(([previous, current]) => previous?.type === 'user' && current?.type === 'anonymous'), // user is logged out and got a new token as an anonymous user
-              switchMap(([previous, current]) =>
-                combineLatest([
-                  store.pipe(select(getLoggedInUser), startWith(undefined), pairwise()),
-                  store.pipe(select(getCurrentBasket)),
-                  store.pipe(select(getSelectedOrderId)),
-                  this.apiToken$,
-                ]).pipe(
-                  take(1),
-                  this.mapToApiTokenCookie(),
-                  filter(calculated => calculated?.type === 'user'), // application calculated an user api token cookie although an anonymous cookie is stored
-                  map<ApiTokenCookie, [ApiTokenCookieType, string]>(() => [previous.type, current.apiToken])
-                )
-              )
-            )
+      )
+    );
+  }
+
+  private logoutRoutine() {
+    this.store
+      .pipe(select(getLoggedInUser), startWith(undefined), pairwise())
+      .pipe(filter(([previous, current]) => !!previous && !current))
+      .subscribe(() => {
+        this.cookiesService.remove('apiToken');
+      });
+  }
+
+  private saveTokenRoutine() {
+    combineLatest([
+      this.store.pipe(select(getLoggedInUser)),
+      this.store.pipe(select(getCurrentBasket)),
+      this.store.pipe(select(getSelectedOrderId)),
+      this.apiToken$,
+    ])
+      .pipe(
+        switchMap(value => of(value)),
+        this.mapToApiTokenCookie(),
+        filter(apiToken => !!apiToken?.apiToken),
+        distinctUntilChanged<ApiTokenCookie>(isEqual)
+      )
+      .subscribe(apiToken => {
+        const cookieContent = apiToken?.apiToken ? JSON.stringify(apiToken) : undefined;
+        if (cookieContent) {
+          this.cookiesService.put('apiToken', cookieContent, {
+            expires: this.cookieOptions?.expires
+              ? new Date(this.cookieOptions.expires.getTime() + 60000)
```

Comment:
why has this been changed? Can you define the 60000 on top as constant like DEFAULT_EXPIRY_TIME